proxy/http_headers.go

```diff
@@ -72,6 +73,20 @@ func addHeaders(r *http.Request, cfg config.Proxy) error {
 	return nil
 }
 
+// addRequestID Set header with name from `cfg.RequestID` to UUID value
+//
+func addRequestID(r *http.Request, p *HTTPProxy) {
+	if p.Config.RequestID != "" {
```

Comment:
Can you add this to the `addHeaders()` function and move your test to the `TestAddHeaders()` fn as well, please? This isn't complex enough for a separate fn/test IMO. 

```go
if p.Config.RequestID != "" {
    id := p.UUID
    if id == nil {
        id = uuid.NewUUID
    }
    r.Header.Set(p.Config.RequestID, id())
}